core/src/main/scala/zio/schema/Schema.scala

```diff
@@ -27,6 +27,410 @@ object Schema {
+    extract: Z => (A1, A2, A3, A4, A5, A6, A7, A8, A9, A10, A11, A12, A13, A14, A15, A16, A17, A18, A19, A20)
+  ) extends CaseClass[Z] {
+    override def toRecord: Record =
+      Record(Map(field1, field2, field3, field4, field5, field6, field7, field8, field9, field10, field11, field12, field13, field14, field15, field16, field17, field18, field19, field20))
+  }
+  case class CaseClass21[A1, A2, A3, A4, A5, A6, A7, A8, A9, A10, A11, A12, A13, A14, A15, A16, A17, A18, A19,A20, A21, Z](
+    field1: (String, Schema[A1]),
+    field2: (String, Schema[A2]),
+    field3: (String, Schema[A3]),
+    field4: (String, Schema[A4]),
+    field5: (String, Schema[A5]),
+    field6: (String, Schema[A6]),
+    field7: (String, Schema[A7]),
+    field8: (String, Schema[A8]),
+    field9: (String, Schema[A9]),
+    field10: (String, Schema[A10]),
+    field11: (String, Schema[A11]),
+    field12: (String, Schema[A12]),
+    field13: (String, Schema[A13]),
+    field14: (String, Schema[A14]),
+    field15: (String, Schema[A15]),
+    field16: (String, Schema[A16]),
+    field17: (String, Schema[A17]),
+    field18: (String, Schema[A18]),
+    field19: (String, Schema[A19]),
+    field20: (String, Schema[A20]),
+    field21: (String, Schema[A21]),
+    construct: (A1, A2, A3, A4, A5, A6, A7, A8, A9, A10, A11, A12, A13, A14, A15, A16, A17, A18, A19, A20, A21) => Z,
+    extract: Z => (A1, A2, A3, A4, A5, A6, A7, A8, A9, A10, A11, A12, A13, A14, A15, A16, A17, A18, A19, A20, A21)
+  ) extends CaseClass[Z] {
+    override def toRecord: Record =
+      Record(Map(field1, field2, field3, field4, field5, field6, field7, field8, field9, field10, field11, field12, field13, field14, field15, field16, field17, field18, field19, field20, field21))
+  }
+  case class CaseClass22[A1, A2, A3, A4, A5, A6, A7, A8, A9, A10, A11, A12, A13, A14, A15, A16, A17, A18, A19, A20, A21, A22, Z](
+    field1: (String, Schema[A1]),
+    field2: (String, Schema[A2]),
+    field3: (String, Schema[A3]),
+    field4: (String, Schema[A4]),
+    field5: (String, Schema[A5]),
+    field6: (String, Schema[A6]),
+    field7: (String, Schema[A7]),
+    field8: (String, Schema[A8]),
+    field9: (String, Schema[A9]),
+    field10: (String, Schema[A10]),
+    field11: (String, Schema[A11]),
+    field12: (String, Schema[A12]),
+    field13: (String, Schema[A13]),
+    field14: (String, Schema[A14]),
+    field15: (String, Schema[A15]),
+    field16: (String, Schema[A16]),
+    field17: (String, Schema[A17]),
+    field18: (String, Schema[A18]),
+    field19: (String, Schema[A19]),
+    field20: (String, Schema[A20]),
+    field21: (String, Schema[A21]),
+    field22: (String, Schema[A22]),
+    construct: (A1, A2, A3, A4, A5, A6, A7, A8, A9, A10, A11, A12, A13, A14, A15, A16, A17, A18, A19, A20, A21, A22) => Z,
+    extract: Z => (A1, A2, A3, A4, A5, A6, A7, A8, A9, A10, A11, A12, A13, A14, A15, A16, A17, A18, A19, A20, A21, A22)
+  ) extends CaseClass[Z] {
+    override def toRecord: Record =
```

Comment:
I love the `toRecord` on all of these, because it gives library authors a convenient way to generically handle all of them, albeit with performance penalty.